### PR TITLE
[Feature] Benchmark storage types

### PR DESCRIPTION
**Pull request by adityagoel4512**
## Description

Adds documentation of collection speed difference when using different storage types in a prototype distributed replay buffer as found in #615. This updates docs and introduces and benchmarks top level directory with the source code for the benchmark.

## Motivation and Context

This addresses the need to document and investigate the sampling latency improvements from the memmap storage over tensor storage and list storage types in distributed RL settings. It validates that it does indeed lead to sampling latency improvements and documents these to encourage users to choose an appropriate storage type.

## Types of changes

What types of changes does your code introduce? Remove all that do not apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation (update in the documentation)
- [ ] Example (update in the folder of examples)

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [ x] I have read the [CONTRIBUTION](https://github.com/pytorch/rl/blob/main/CONTRIBUTING.md) guide (**required**)
- [ x] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [x ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by adityagoel4512** [comment]
I'd like to wait for #615 to be resolved before attempting to merge this PR

**Comment by adityagoel4512** [comment]
#615 is merged now!

File: benchmarks/storage/benchmark_sample_latency_over_rpc.py
Comment (vmoens) [inline benchmarks/storage/benchmark_sample_latency_over_rpc.py:5]: Can you provide a simple example on how to run that script (since you need to run it twice with different ranks)?

File: docs/source/reference/data.rst
Comment (vmoens) [inline docs/source/reference/data.rst:59]: wonderful!

**Review by vmoens — APPROVED** [review]
LGTM thanks, it's marvellous 
Can you just add a small comment at the beginning of the script explaining how to run it?

File: benchmarks/storage/benchmark_sample_latency_over_rpc.py
Comment (adityagoel4512) [reply benchmarks/storage/benchmark_sample_latency_over_rpc.py:9]: Hopefully this works

**Pull request merged**
